### PR TITLE
feat: add new task for fevm trace

### PR DESCRIPTION
**Pull request by Terryhung**

### CODE REVIEW

File: tasks/fevm/vm/task.go [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+				From:            child.Message.From.String(),
+				FromEthAddress:  fromEthAddress,
+				ToEthAddress:    toEthAddress,
+				Value:           child.Message.Value.String(),
+				GasUsed:         0,
+				ExitCode:        int64(child.Receipt.ExitCode),
+				ActorCode:       toActorCode,
+				Method:          uint64(child.Message.Method),
+				Index:           child.Index,
+				Params:          base64.StdEncoding.EncodeToString(child.Message.Params),
```
Comment (Terryhung) [inline tasks/fevm/vm/task.go:164]: Referring to the params value returned by `lily chain getmessage`, we also encode params value in Base64.

Comment (birdychang) [reply tasks/fevm/vm/task.go:168]: what's the benefit to encode in base64?

File: model/fevm/vm.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import (
+	"context"
+
+	"go.opencensus.io/tag"
+
+	"github.com/filecoin-project/lily/metrics"
+	"github.com/filecoin-project/lily/model"
+)
+
+type FEVMVMMessage struct {
```
[the remark this answers is not in the document]
Comment (Terryhung) [reply model/fevm/vm.go:12]: Agreed!

File: tasks/fevm/vm/task.go (hunk 1 above)
Comment (Terryhung) [reply tasks/fevm/vm/task.go:168]: I think the one of reasons for encoding `params` into base64 is to ensure compatibility: Base64 encoding is a widely supported format, it is able to to mitigate risks across different operating systems and databases.

File: tasks/fevm/trace/task.go [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+				FromEthAddress:   fromEthAddress,
+				ToEthAddress:     toEthAddress,
+				Value:            child.Message.Value.String(),
+				GasUsed:          0,
+				ExitCode:         int64(child.Receipt.ExitCode),
+				ActorCode:        toActorCode,
+				Method:           uint64(child.Message.Method),
+				Index:            child.Index,
+				Params:           base64.StdEncoding.EncodeToString(child.Message.Params),
+				Returns:          base64.StdEncoding.EncodeToString(child.Receipt.Return),
```
Comment (birdychang) [inline tasks/fevm/trace/task.go:169]: maybe use `EthBytes` to be consistent with eth?

File: tasks/fevm/trace/task.go [hunk 4]
```diff
@@ -0,0 +1,199 @@
+package fevmvm
```
[the remark this answers is not in the document]
Comment (Terryhung) [reply tasks/fevm/trace/task.go:1]: fixed

File: tasks/fevm/trace/task.go [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+	builtintypes "github.com/filecoin-project/go-state-types/builtin"
+	"github.com/filecoin-project/lily/lens"
+	"github.com/filecoin-project/lily/lens/util"
+	"github.com/filecoin-project/lily/model"
+	"github.com/filecoin-project/lily/model/fevm"
+	visormodel "github.com/filecoin-project/lily/model/visor"
+	tasks "github.com/filecoin-project/lily/tasks"
+)
+
+var log = logging.Logger("lily/tasks/fevmvm")
```
[the remark this answers is not in the document]
Comment (Terryhung) [reply tasks/fevm/trace/task.go:27]: fixed

File: tasks/fevm/trace/task.go (hunk 3 above)
Comment (Terryhung) [reply tasks/fevm/trace/task.go:169]: changed

File: schemas/v1/26_fevm_traces.go [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+func init() {
+	patches.Register(
+		26,
+		`
+ CREATE TABLE IF NOT EXISTS {{ .SchemaName | default "public"}}.fevm_traces (
+    height BIGINT NOT NULL,
+    message_state_root TEXT,
+    transaction_hash TEXT,
+    message_cid TEXT,
+	trace_cid TEXT,
```
[the remark this answers is not in the document]
Comment (Terryhung) [reply schemas/v1/26_fevm_traces.go:12]: fixed

File: tasks/fevm/trace/task.go [hunk 7]
```diff
@@ -0,0 +1,199 @@
...
+				TraceCid:         getMessageTraceCid(child.Message).String(),
+				To:               child.Message.To.String(),
+				From:             child.Message.From.String(),
+				FromEthAddress:   fromEthAddress,
+				ToEthAddress:     toEthAddress,
+				Value:            child.Message.Value.String(),
+				GasUsed:          0,
+				ExitCode:         int64(child.Receipt.ExitCode),
+				ActorCode:        toActorCode,
+				Method:           uint64(child.Message.Method),
```
Comment (Terryhung) [inline tasks/fevm/trace/task.go:166]: try to store readable method name

File: tasks/fevm/trace/task.go [hunk 8]
```diff
@@ -0,0 +1,199 @@
...
+				TransactionHash:  transaction.Hash.String(),
+				MessageStateRoot: parentMsg.StateRoot.String(),
+				MessageCid:       parentMsg.Cid.String(),
+				TraceCid:         getMessageTraceCid(child.Message).String(),
+				To:               child.Message.To.String(),
+				From:             child.Message.From.String(),
+				FromEthAddress:   fromEthAddress,
+				ToEthAddress:     toEthAddress,
+				Value:            child.Message.Value.String(),
+				GasUsed:          0,
```
Comment (Terryhung) [inline tasks/fevm/trace/task.go:163]: remove gas_used

**Review by birdychang — APPROVED** [review]

**Pull request merged**